jack/pcm_jack.c

```diff
@@ -147,35 +147,38 @@ snd_pcm_jack_process_cb(jack_nframes_t nframes, snd_pcm_ioplug_t *io)
 		jack->areas[channel].first = 0;
 		jack->areas[channel].step = jack->sample_bits;
 	}
-		
-	if (io->state != SND_PCM_STATE_RUNNING) {
-		if (io->stream == SND_PCM_STREAM_PLAYBACK) {
-			for (channel = 0; channel < io->channels; channel++)
-				snd_pcm_area_silence(&jack->areas[channel], 0, nframes, io->format);
-			return 0;
-		}
-	}
-	
-	areas = snd_pcm_ioplug_mmap_areas(io);
 
-	while (xfer < nframes) {
-		snd_pcm_uframes_t frames = nframes - xfer;
-		snd_pcm_uframes_t offset = jack->hw_ptr;
-		snd_pcm_uframes_t cont = io->buffer_size - offset;
+	if (io->state == SND_PCM_STATE_RUNNING) {
+		areas = snd_pcm_ioplug_mmap_areas(io);
+
+		while (xfer < nframes) {
```

Comment:
where is the improvement? Doesn't this still copy nframes w/o checking availability ?

Reply:
I will change the commit message.
With this commit the JACK buffer will only be filled with silence in case this plugin is not in RUNNING state e.g. in PREPARED or XRUN